server/utils/httputils/utils.go

```diff
@@ -81,17 +67,55 @@ func WriteJSONStatus(w http.ResponseWriter, statusCode int, v interface{}) {
 }
 
 func WriteInternalServerError(w http.ResponseWriter, err error) {
-	WriteJSONError(w, http.StatusInternalServerError, "An internal error has occurred. Check app server logs for details.", err)
+	WriteError(w, http.StatusInternalServerError, err)
 }
 
 func WriteBadRequestError(w http.ResponseWriter, err error) {
-	WriteJSONError(w, http.StatusBadRequest, "Invalid request.", err)
+	WriteError(w, http.StatusBadRequest, err)
 }
 
 func WriteNotFoundError(w http.ResponseWriter, err error) {
-	WriteJSONError(w, http.StatusNotFound, "Not found.", err)
+	WriteError(w, http.StatusNotFound, err)
 }
 
 func WriteUnauthorizedError(w http.ResponseWriter, err error) {
-	WriteJSONError(w, http.StatusUnauthorized, "Unauthorized.", err)
+	WriteError(w, http.StatusUnauthorized, err)
+}
+
+const InLimit = 10 * (1 << 20)
+
+func ReadAndClose(in io.ReadCloser) ([]byte, error) {
+	defer in.Close()
+	return LimitReadAll(in, InLimit)
+}
+
+func LimitReadAll(in io.Reader, limit int64) ([]byte, error) {
+	if in == nil {
+		return []byte{}, nil
+	}
+	return ioutil.ReadAll(&io.LimitedReader{R: in, N: limit})
+}
+
+func ProcessResponseError(w http.ResponseWriter, resp *http.Response, err error) bool {
+	if err != nil {
+		WriteError(w, http.StatusInternalServerError, err)
+		return true
+	}
+	if resp.StatusCode != http.StatusOK {
+		bb, _ := ReadAndClose(resp.Body)
+		WriteError(w, http.StatusBadGateway,
+			errors.Errorf("received status %v: %s", resp.Status, string(bb)))
+		return true
+	}
+	return false
+}
+
+func GetFromURL(url string) ([]byte, error) {
+	// nolint:gosec
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	return ioutil.ReadAll(resp.Body)
```

Comment:
```suggestion
	return io.ReadAll(resp.Body)
```

Reply:
`ReadAll` is not in the `io` package, or am I behind on my go versions?

Reply:
Yes, go 1.16 added it https://golang.org/pkg/io/#ReadAll

Reply:
👍  already got it, thx!